### PR TITLE
Chore/provision nexus

### PR DESCRIPTION
**Pull request by binhoul**

### CODE REVIEW

File: modules/nexus/curator.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module "curator" {
+  source = "../curator/action"
+
+  key               = "nexus"
+  disable           = "${var.curator_enable ? "false" : "true"}"
```
[the remark this answers is not in the document]
Comment (lawliet89) [reply modules/nexus/curator.tf]: In Terraform 0.11, boolean converts to 0 or 1.

File: modules/nexus/curator.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@ module "curator" {
   source = "../curator/action"
 
   key               = "nexus"
-  disable           = "${var.curator_enable ? "false" : "true"}"
+  disable           = "${var.curator_enabled ? 0 : 1}"
```
Comment (lawliet89) [inline modules/nexus/curator.tf]: I mean you should have left the line as before. Now it probably won't work.

File: modules/nexus/curator.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (binhoul) [reply modules/nexus/curator.tf]: understand.

File: roles/nexus3/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    checksum: "md5:e7a6b33deb17c66d46beb42f1b8cf3bc"
+
+- name: unpack java 8 package
+  unarchive:
+    src: "/tmp/{{ java_package_name }}"
+    dest: "/usr/local"
+    remote_src: yes
+  become: yes
+
+- name: symblic link java binary file to /usr/bin
```
Comment (guangie88) [inline roles/nexus3/tasks/main.yml]: Nitpick: `symblic` -> `symbolic`

File: modules/nexus/packer/ami/packer.json [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                "-e",
+                "ca_certificate={{user `ca_certificate`}}",
+                "-e",
+                "consul_host={{user `consul_host`}} consul_port={{user `consul_port`}} consul_scheme={{user `consul_scheme`}} consul_token={{user `consul_token`}} consul_integration_prefix={{user `consul_integration_prefix`}}",
+                "-e",
+                "nexus_version={{user `nexus_version`}} nexus_db_dir={{user `nexus_db_dir`}}",
+                "-e",
+                "timezone={{user `timezone`}}",
+                "-e",
+                "ansible_python_interpreter=/usr/bin/python3"
```
Comment (guangie88) [inline modules/nexus/packer/ami/packer.json]: Guess you could try this too? https://github.com/datagovsg/terraform-modules/pull/233/files#diff-6ab1fec7168f06f35842e38b6ae11cf9R99

**Comment by guangie88** [comment]
Noob qn: Why build for AMI instead of Nomad job?

**Comment by binhoul** [comment]
> Noob qn: Why build for AMI instead of Nomad job?

We consider this as a seperated module, not belonging to the nomad cluster, and another reason it is stateful.

**Pull request merged**